### PR TITLE
Add support for payment method snapshots

### PR DESCRIPTION
**Pull request by sslotnick**
New PaymentMethodSnapshot object returned by Zuora which contains a snapshot of the payment method at the time of a specific payment

### CODE REVIEW

File: lib/zuora/objects/base.rb [hunk 1]
```diff
@@ -140,6 +144,7 @@ def update
 
     # destroy the remote object
     def destroy
```
Comment (adamjk) [inline lib/zuora/objects/base.rb]: are these destroy semantics being pulled in from somewhere else?

Comment (sslotnick) [inline lib/zuora/objects/base.rb]: They are copied from ActiveRecord. I added a comment that describes why these methods are needed.

File: lib/zuora/objects/payment_method_snapshot.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
+module Zuora::Objects
```
Comment (adamjk) [inline lib/zuora/objects/payment_method_snapshot.rb]: so basically this is just a copy of payment_method.rb for the account?

Comment (sslotnick) [inline lib/zuora/objects/payment_method_snapshot.rb]: For the payment, yes. The payment has 2 objects associated with it: a pointer to the payment method in its current state and a snapshot of the payment method as it existed when the payment was processed/attempted.

**Pull request merged**